### PR TITLE
#KTOR-936 Migrate to explicit api mode(part 1)

### PR DESCRIPTION
**Pull request by e5l**
Fixes [`#KTOR-936`](https://youtrack.jetbrains.com/issue/KTOR-936)

### CODE REVIEW

File: ktor-io/jvm/src/io/ktor/utils/io/ByteBufferChannel.kt [hunk 1]
```diff
@@ -27,7 +29,7 @@ internal open class ByteBufferChannel(
 ) : ByteChannel, ByteReadChannel, ByteWriteChannel, LookAheadSuspendSession, HasReadSession, HasWriteSession {
 
     // internal constructor for reading of byte buffers
```
Comment (cy6erGn0m) [inline ktor-io/jvm/src/io/ktor/utils/io/ByteBufferChannel.kt:31]: Looks like it is not true anymore. We could remove the comment.

**Review by cy6erGn0m — APPROVED** [review]

File: gradle.properties [hunk 2]
```diff
@@ -2,7 +2,7 @@
 kotlin.code.style=official
 
 # config
-version=1.4.0
+version=1.4.1-SNAPSHOT
```
Comment (Stexxe) [inline gradle.properties:5]: Most likely commit "KTOR-1027 Update versions" shouldn't be here

File: ktor-io/common/src/io/ktor/utils/io/core/AbstractInput.kt [hunk 3]
```diff
@@ -314,13 +314,13 @@ abstract class AbstractInput(
     }
 
     @Deprecated("Binary compatibility.", level = DeprecationLevel.HIDDEN)
-    final override fun readShort() = readShort()
+    final override fun readShort(): Short = readShort()
```
Comment (Stexxe) [inline ktor-io/common/src/io/ktor/utils/io/core/AbstractInput.kt:317]: Should `public` modifier be here?

**Review by Stexxe — CHANGES_REQUESTED** [review]

File: ktor-io/common/src/io/ktor/utils/io/core/AbstractInput.kt (hunk 3 above)
Comment (e5l) [reply ktor-io/common/src/io/ktor/utils/io/core/AbstractInput.kt:317]: Nope. We should set visibility modifiers for `overrides` only when it changes visibility

File: gradle.properties (hunk 2 above)
Comment (e5l) [reply gradle.properties:5]: rebased

File: ktor-io/jvm/src/io/ktor/utils/io/ByteBufferChannel.kt (hunk 1 above)
Comment (e5l) [reply ktor-io/jvm/src/io/ktor/utils/io/ByteBufferChannel.kt:29]: thanks, done

**Review by Stexxe — APPROVED** [review]

**Pull request closed without merging**